Framework/Core/src/DispatcherDPL.cxx

```diff
@@ -51,7 +51,7 @@ void DispatcherDPL::processCallback(ProcessingContext& ctx, BernoulliGenerator&
         ctx.outputs().adopt(outputSpec, DataRefUtils::as<TObject>(input).release());
       } else { // POD
         // todo: use API for that when it is available
-        ctx.outputs().adoptChunk(outputSpec, const_cast<char*>(input.payload), inputHeader->size(),
+        ctx.outputs().adoptChunk(outputSpec, const_cast<char*>(input.payload), inputHeader->payloadSize,
```

Comment:
ah, nice, this is interesting. So it means that the `DataAllocator` is aware of the chunk being data from the input. Is this correct, @ktf?

Reply:
IIRC the sending happens before the old inputs are removed. So this should be safe. We should probably document this is a guarantee.

Reply:
but adopting means to transfer ownership, so the `DataAllocator` also might want to clean up?

Reply:
yes, but IIRC, it works with a `unique_ptr` which gets moved. Probably what breaks is that afterwards you cannot refer to the input safely anymore, but actual memory management should be fine. Anyways, to be checked.